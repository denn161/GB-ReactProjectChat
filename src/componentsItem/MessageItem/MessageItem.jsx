
import PropTypes from 'prop-types';

import Message from './Message';
import styles from './MessageItem.module.css';

const MessageItem = (props) => {
  const { messages = [], removeMessage } = props
  
  if(!messages.length) return null
            
  return (
    <>
      <ul className={styles.messages__list}>
    {messages.map((message) =>
      <Message key={message.id} message={message}
        removeMessage={removeMessage} />

        )      
        
        }
</ul>
      
      </>
  );
}

MessageItem.propTypes = {
  messages: PropTypes.array,
  removeMessage:PropTypes.func
}
export default MessageItem;