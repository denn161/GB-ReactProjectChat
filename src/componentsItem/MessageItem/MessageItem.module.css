
.messages__list{
  list-style: none;
}