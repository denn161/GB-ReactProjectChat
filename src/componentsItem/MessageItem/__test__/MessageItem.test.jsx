
import { render, screen } from '@testing-library/react';
import MessageItem from '../MessageItem';


describe('MessageItem component', () => {
    
    it('MessageiItem renders', () => {
        
  const data = [{ id: 1, name: 'User1', message: 'hello friend' },
                { id: 2, name: 'User2', message: 'hello friend1' },
                { id: 3, name: 'User3', message: 'hello friend2' }]

    render(<MessageItem messages={data} />)
        
        const userName = screen.getByText('User1')
        const userMsg = screen.getByText('hello friend')

        expect(screen.getByRole('list')).toBeInTheDocument()
        expect(userName).toBeInTheDocument()  
        expect(userMsg).toBeInTheDocument()
        expect(screen.getByText('User2')).toBeInTheDocument()
        expect(screen.getByText('User3')).toBeInTheDocument()
        expect(screen.getByText('hello friend1')).toBeInTheDocument()
        expect(screen.getByText('hello friend2')).toBeInTheDocument()        
        expect(data).toContainEqual({id:1,name:'User1',message:'hello friend'}
        )      

    }) 

//    it('MessageiItem renders without data', () => {
  
//       render(<MessageItem />)
          
//       expect(screen.queryByRole('list')).toBeNull()
       
//     })
    
    it('MessageiItem snapshot', () => {
        
  const data = [{ id: 1, name: 'User1', message: 'hello friend' },
                { id: 2, name: 'User2', message: 'hello friend1' },
                { id: 3, name: 'User3', message: 'hello friend2' }]        
  const messages = render(<MessageItem messages={data} />)
        
  expect(messages).toMatchSnapshot()           
       
    })
    it('MessageiItem empty snapshot ', () => {        
      
       const messages = render(<MessageItem/>)
        
       expect(messages).toMatchSnapshot()           
       
    })
})




