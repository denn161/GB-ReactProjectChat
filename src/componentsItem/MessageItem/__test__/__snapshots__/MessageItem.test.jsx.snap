// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageItem component MessageiItem empty snapshot  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`MessageItem component MessageiItem snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="messages__list"
      >
        <li
          class="item"
          style="margin: 10px 10px 10px auto; border: 2px solid green; padding: 20px; border-radius: 10px;"
        >
          <span>
            User1
          </span>
          <p>
            hello friend
          </p>
          <img
            alt="close"
            class="btn"
            src="close.png"
          />
        </li>
        <li
          class="item"
          style="margin: 10px 10px 10px auto; border: 2px solid green; padding: 20px; border-radius: 10px;"
        >
          <span>
            User2
          </span>
          <p>
            hello friend1
          </p>
          <img
            alt="close"
            class="btn"
            src="close.png"
          />
        </li>
        <li
          class="item"
          style="margin: 10px 10px 10px auto; border: 2px solid green; padding: 20px; border-radius: 10px;"
        >
          <span>
            User3
          </span>
          <p>
            hello friend2
          </p>
          <img
            alt="close"
            class="btn"
            src="close.png"
          />
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="messages__list"
    >
      <li
        class="item"
        style="margin: 10px 10px 10px auto; border: 2px solid green; padding: 20px; border-radius: 10px;"
      >
        <span>
          User1
        </span>
        <p>
          hello friend
        </p>
        <img
          alt="close"
          class="btn"
          src="close.png"
        />
      </li>
      <li
        class="item"
        style="margin: 10px 10px 10px auto; border: 2px solid green; padding: 20px; border-radius: 10px;"
      >
        <span>
          User2
        </span>
        <p>
          hello friend1
        </p>
        <img
          alt="close"
          class="btn"
          src="close.png"
        />
      </li>
      <li
        class="item"
        style="margin: 10px 10px 10px auto; border: 2px solid green; padding: 20px; border-radius: 10px;"
      >
        <span>
          User3
        </span>
        <p>
          hello friend2
        </p>
        <img
          alt="close"
          class="btn"
          src="close.png"
        />
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
